layer/layer_data.cc

```diff
@@ -90,6 +90,14 @@ void LayerData::Log(const std::vector<uint64_t>& pipeline,
   fflush(out_);
 }
 
+void LayerData::Log(const std::vector<uint64_t>& pipeline,
+                    const char* str) const {
```

Comment:
What do you think about just passing a std::string instead and using something less brittle than `snprintf` at the call sites, e.g., std::stringstream or absl::StrCat?

Reply:
Done